### ISSUE TITLE
OpenBSD 6.3 build: test/arith_uint256_tests.cpp complains about missing argument

### ISSUE THREAD

**Issue by grim-trigger** [bitcoin/bitcoin#13337]
When running `gmake check`, test/arith_uint256_tests.cpp complains about missing argument.

Expected successful `gmake check` results.

Output from bitcoin/src/test/arith_uint256_tests.cpp.log:

```
Missing an argument value for the parameter run_test in the argument 

Parameter: run_test
 Filters, which test units to include or exclude from test module execution.
 Command line formats:
   --run_test=<test unit filter>
   -t <test unit filter>
 Environment variable: BOOST_TEST_RUN_FILTERS

For detailed help on Boost.Test parameters use:
  test_bitcoin --help
or
  test_bitcoin --help=<parameter name>
```

This issue occurs when running `gmake check`.

This issue occurs with bitcoin master version 610f4dd7

This issue occurs with OpenBSD 6.3 AMD64. 
File [config.log](https://github.com/bitcoin/bitcoin/files/2045192/config.log) attached.

**Comment by laanwj** [comment]
Current 0.16 branch and master pass `gmake check` on OpenBSD 5.2, here.
So interestingly, this looks like a 5.3 specific problem? Or does this depend on the default shell? (`/bin/ksh` in my case)

**Comment by practicalswift** [comment]
I'm able to reproduce this issue under stock OpenBSD 6.3 using the default shell:

```
$ uname -r
6.3
$ echo $0
-ksh
$ gmake check
Making check in src
gmake[1]: Entering directory '/.../bitcoin/src'
gmake[2]: Entering directory '/.../bitcoin/src'
gmake[3]: Entering directory '/.../bitcoin'
gmake[3]: Leaving directory '/.../bitcoin'
gmake  check-TESTS check-local
gmake[3]: Entering directory '/.../bitcoin/src'
gmake[4]: Entering directory '/.../bitcoin/src'
============================================================================
Testsuite summary for Bitcoin Core 0.16.99
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
gmake[4]: Leaving directory '/.../bitcoin/src'
Running tests: from test/arith_uint256_tests.cpp
Missing an argument value for the parameter run_test in the argument

Parameter: run_test
 Filters, which test units to include or exclude from test module execution.
 Command line formats:
   --run_test=<test unit filter>
   -t <test unit filter>
 Environment variable: BOOST_TEST_RUN_FILTERS

For detailed help on Boost.Test parameters use:
  test_bitcoin --help
or
  test_bitcoin --help=<parameter name>
gmake[3]: *** [Makefile:10247: test/arith_uint256_tests.cpp.test] Error 1
gmake[3]: Leaving directory '/.../bitcoin/src'
gmake[2]: *** [Makefile:9996: check-am] Error 2
gmake[2]: Leaving directory '/.../bitcoin/src'
gmake[1]: *** [Makefile:9684: check-recursive] Error 1
gmake[1]: Leaving directory '/.../bitcoin/src'
gmake: *** [Makefile:754: check-recursive] Error 1
```

**Comment by practicalswift** [comment]
GNU grep and BSD grep differs in their handling of regexps when extended regular expressions are not enabled via the `-E` flag:

```
$ grep --version | head -1
grep (GNU grep) 3.1
$ echo "BOOST_FIXTURE_TEST_SUITE(foo)" | grep "BOOST_FIXTURE_TEST_SUITE(\|BOOST_AUTO_TEST_SUITE("
BOOST_FIXTURE_TEST_SUITE(foo)
$
```

```
$ grep --version | head -1
grep version 0.9
$ echo "BOOST_FIXTURE_TEST_SUITE(foo)" | grep "BOOST_FIXTURE_TEST_SUITE(\|BOOST_AUTO_TEST_SUITE("
$ 
```

The portable way to do it is:

```
$ echo "BOOST_FIXTURE_TEST_SUITE(foo)" | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()"
BOOST_FIXTURE_TEST_SUITE(foo)
$
```

I'll submit a fix!

**Comment by practicalswift** [comment]
Fix submitted in #13355.

**Comment by grim-trigger** [comment]
I can confirm that #13355 resolves this issue. Thank you!
